### DISCUSSION TITLE
Can you share sessions between the Server and the Webserver?

### DISCUSSION THREAD

**Discussion by gdsmith** [serverpod/serverpod#681] [Q&A] [answered]
If we set session data on the webserver will that be available on the server?

**Comment by yahu1031** [comment]
Checkout #642

**Reply by gdsmith** [reply]
I'm more looking at sharing session data between Widgets and Endpoints rather than code sharing between client and server.

**Comment by vlidholt** [comment, marked as answer]
Hi @gdsmith, sorry for being a bit slow here. The session objects that you use on the webserver are the same as in any API call. So you have access to the exact same data. The session objects themselves will be different though (but a new object is created for each API call, so it's the same).

**Reply by gdsmith** [reply]
I think I may have been a bit confused about sessions in ServerPod and how everything fits together. Essentially I am trying to save data per client between requests. Is it possible to save that as session data between endpoint requests? Is there any way to share that data between the widgets and the endpoints?

**Reply by vlidholt** [reply]
Yes, typically you will want to use the database to save this data.

It is also possible to save data in the memory of the server in Serverpod (e.g. through a global variable). However, if you do this, you will need to be mindful as that data will disappear when the server restarts. If you are running your application on multiple servers you will need to figure out how to share that state too.

**Comment by gdsmith** [comment]
I was trying to use session cookies to identify the client but the client doesn't seem to save and send them on subsequent calls. Is there a better way to identify the client?

**Reply by vlidholt** [reply]
If you want to sign in users, then use the auth package: https://docs.serverpod.dev/concepts/authentication
If you want to store data on the client, then the shared_preferences package is a good option: https://pub.dev/packages/shared_preferences
